### ISSUE TITLE
Consider disabling precompute epoch on overloaded nodes

### ISSUE THREAD

**Issue by dapplion** [ChainSafe/lodestar#3513]
**Describe the bug**

In contabo-12 2021-12-12 08:02:13 CET the node was very overloaded due to high memory usage with GC using >75% of CPU time. In that situation everything is extremely slow. The precompute epoch kicked in and ran an epoch transition which according to metrics had a total async time of 36 sec. 

![Screenshot from 2021-12-12 12-01-39](https://user-images.githubusercontent.com/35266934/145709838-f27c9aa6-c970-44d5-88b2-77a7a44522d6.png)

The resulting state is guaranteed to be useless. However it does not show up in the "Precompute Epoch Transition" "Hit vs Miss" chart

![Screenshot from 2021-12-12 12-06-33](https://user-images.githubusercontent.com/35266934/145709865-765fc1aa-b05f-45ed-80f7-d6cc991953b8.png)

**Expected behavior**

- Consider disabling precompute epoch on overloaded nodes. This item is not a priority, since we should work on preventing the node from getting overloaded in the first place.
- Review metrics, the state above could not have been used but it still didn't show up as wasted. Maybe I'm missing something?

Also the charts could be improved, it's very hard to get a picture of what's going on with the way the info is provided now. For example the chart "Run Result"

![Screenshot from 2021-12-12 12-14-19](https://user-images.githubusercontent.com/35266934/145710077-a3e7bbe1-f668-4376-a720-9d89f01daf27.png)

Is answering the question "What's it the run result of each epoch?, skipped or done? In my opinion doing this tweaks it looks much better
- Scale up to rate per epoch
- Stack
- Reduce interval rate

![Screenshot from 2021-12-12 12-11-06](https://user-images.githubusercontent.com/35266934/145710031-067973ee-0ad1-436d-bc09-6bd22a051bca.png)

Now you can clearly see that:
- exactly once run is done per epoch
- the % of skipped runs

**Comment by twoeths** [comment]
@dapplion  The precomputeNextEpochTransition only gets triggered when `headSlot === clockSlot` so I'm not sure why it's useless, since when the next gossip block comes we'll run the epoch transition anyway

regarding the chart, agree that it looks much better that way 👍

**Comment by dapplion** [comment]
> The precomputeNextEpochTransition only gets triggered when headSlot === clockSlot

Good point! Then could it be that the node is so overloaded that the clock is also behind?

**Comment by twoeths** [comment]
> Good point! Then could it be that the node is so overloaded that the clock is also behind?

@dapplion due to #3517 contabo-12 was so unlucky to get  connected to some Unknown peers that don't support any protocols and it caused this memory issue. I'd like to continue monitoring this for a while to decide if precomputing epoch transition is an issue or not.

**Comment by dapplion** [comment]
Closing as it has not been observed for months. If the node is really overloaded there are other areas of greater concern that this.
